shell/platform/darwin/ios/ios_external_view_embedder.h

```diff
@@ -0,0 +1,71 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_DARWIN_IOS_IOS_EXTERNAL_VIEW_EMBEDDER_H_
+#define FLUTTER_SHELL_PLATFORM_DARWIN_IOS_IOS_EXTERNAL_VIEW_EMBEDDER_H_
+
+#include "flutter/flow/embedded_views.h"
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h"
+
+namespace flutter {
+
+class IOSExternalViewEmbedder : public ExternalViewEmbedder {
+ public:
+  IOSExternalViewEmbedder(
+      FlutterPlatformViewsController* platform_views_controller,
```

Comment:
What are the ownership rules and lifecycle for `FlutterPlatformViewsController`?

Reply:
Can `platform_views_controller` ever be `nullptr`?

Reply:
I attempted to add a check here, but in some cases we do seem to create with `nullptr`, see: https://github.com/flutter/engine/blob/master/shell/platform/darwin/ios/framework/Source/FlutterOverlayView.mm#L69.

Reply:
Coming to the ownership question, looks like `FlutterEngine.mm` manages the lifecycle of the `FlutterPlatformViewsController`, I don't like that this is a raw pointer that we pass around, I'd like it to be a shared pointer. Mind if I file an issue for this and address it as a follow up commit?